src/sentry_core.c

```diff
@@ -385,6 +385,45 @@ sentry_capture_event(sentry_value_t event)
     return was_captured ? event_id : sentry_uuid_nil();
 }
 
+bool
+sentry__make_sampling_decision(double sample_rate)
+{
+    uint64_t rnd;
+    return sample_rate < 1.0 && !sentry__getrandom(&rnd, sizeof(rnd))
```

Comment:
Why does this return `false` when `sample_rate ≥ 1.0`? I would interpret that as "everything should be sampled", shouldn't it always return `true` in that case?

Reply:
thanks for catching this, looks like i messed up that scenario.

Reply:
I wouldn't really expect a method called `is_unsampled` to depend on a random number.

Reply:
I think `is_unsampled` is not a good name either; you'd expect a function with that name to tell you whether a thing passed to it is unsampled, but its only argument is the sampling rate. Really what this function does is return `true` with a probability of `sample_rate`. I would suggest a name like `sentry__random_bool`.

Reply:
it is now `sentry__roll_dice` 🎉 this isn't a part of the public API, so this name shouldn't kick up too much of a fuss.